File: app/src/main/java/com/example/nik/mixology/Activities/ActivitySearch.java
package com.example.nik.mixology.Activities;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;

import com.example.nik.mixology.Adapters.SearchAdapter;
import com.example.nik.mixology.Fragments.FragmentDetails;
import com.example.nik.mixology.Model.Cocktail;
import com.example.nik.mixology.R;

public class ActivitySearch extends AppCompatActivity implements SearchAdapter.OnAdapterItemSelectedListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_search);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);

        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        String query = getIntent().getStringExtra("Query");
        String queryAdjusted = query.replaceAll("%20"," ");
        getSupportActionBar().setTitle(queryAdjusted);


    }

    @Override
    public void onItemSelected(Cocktail id) {
        FragmentDetails detailsFragment = (FragmentDetails) getSupportFragmentManager().findFragmentById(R.id.fragment_details);
        if (detailsFragment == null) {
            Intent mCocktailDetailIntent = new Intent(this, ActivityDetails.class);
            mCocktailDetailIntent.putExtra("Cocktail", id);
            startActivity(mCocktailDetailIntent);
        }
    }
}

File: app/src/main/java/com/capstone/nik/mixology/Activities/ActivitySearch.kt
package com.capstone.nik.mixology.Activities

import android.content.Intent
import android.os.Bundle
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import com.capstone.nik.mixology.R
import com.capstone.nik.mixology.ui.search.SearchRoute
import com.capstone.nik.mixology.ui.theme.MixologyTheme

class ActivitySearch : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val query = intent.getStringExtra(getString(R.string.intent_search_intent_query)).orEmpty()
        enableEdgeToEdge()
        setContent {
            MixologyTheme {
                SearchRoute(
                    initialQuery = query,
                    onBack = { finish() },
                    onDrinkClick = { cocktail ->
                        startActivity(
                            Intent(this, ActivityDetails::class.java)
                                .putExtra(getString(R.string.intent_details_intent_cocktail), cocktail),
                        )
                    },
                )
            }
        }
    }
}
